pkg/services/ngalert/notifier/alertmanager_test.go

```diff
@@ -53,367 +45,7 @@ func setupAMTest(t *testing.T) *Alertmanager {
-
-			iter := am.alerts.GetPending()
-			defer iter.Close()
-			for a := range iter.Next() {
-				alerts = append(alerts, a)
-			}
-
-			// We take the "now" time from one of the UpdatedAt.
-			now := alerts[0].UpdatedAt
-			expAlerts := c.expAlerts(now)
-
-			sort.Sort(types.AlertSlice(expAlerts))
-			sort.Sort(types.AlertSlice(alerts))
-
-			require.Equal(t, expAlerts, alerts)
-		})
-	}
-}
-
-// Tests cleanup of expired Silences. We rely on prometheus/alertmanager for
-// our alert silencing functionality, so we rely on its tests. However, we
-// implement a custom maintenance function for silences, because we snapshot
-// our data differently, so we test that functionality.
-func TestSilenceCleanup(t *testing.T) {
-	require := require.New(t)
-
-	oldRetention := retentionNotificationsAndSilences
-	retentionNotificationsAndSilences = 30 * time.Millisecond
-	oldMaintenance := silenceMaintenanceInterval
-	silenceMaintenanceInterval = 15 * time.Millisecond
-	t.Cleanup(
-		func() {
-			retentionNotificationsAndSilences = oldRetention
-			silenceMaintenanceInterval = oldMaintenance
-		})
-
-	am := setupAMTest(t)
-	now := time.Now()
-	dt := func(t time.Time) strfmt.DateTime { return strfmt.DateTime(t) }
-
-	makeSilence := func(comment string, createdBy string,
-		startsAt, endsAt strfmt.DateTime, matchers models.Matchers) *apimodels.PostableSilence {
-		return &apimodels.PostableSilence{
-			ID: "",
-			Silence: models.Silence{
-				Comment:   &comment,
-				CreatedBy: &createdBy,
-				StartsAt:  &startsAt,
-				EndsAt:    &endsAt,
-				Matchers:  matchers,
-			},
-		}
-	}
-
-	tru := true
-	testString := "testName"
-	matchers := models.Matchers{&models.Matcher{Name: &testString, IsEqual: &tru, IsRegex: &tru, Value: &testString}}
-	// Create silences - one in the future, one currently active, one expired but
-	// retained, one expired and not retained.
-	silences := []*apimodels.PostableSilence{
-		// Active in future
-		makeSilence("", "tests", dt(now.Add(5*time.Hour)), dt(now.Add(6*time.Hour)), matchers),
-		// Active now
-		makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(6*time.Hour)), matchers),
-		// Expiring soon.
-		makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(2*time.Second)), matchers),
-		// Expiring *very* soon
-		makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(1*time.Second)), matchers),
-	}
-
-	for _, s := range silences {
-		_, err := am.CreateSilence(s)
-		require.NoError(err)
-	}
-
-	// Let enough time pass for the maintenance window to run.
-	require.Eventually(func() bool {
-		// So, what silences do we have now?
-		found, err := am.ListSilences(nil)
-		require.NoError(err)
-		return len(found) == 3
-	}, 3*time.Second, 100*time.Millisecond)
-
-	// Wait again for another silence to expire.
-	require.Eventually(func() bool {
-		found, err := am.ListSilences(nil)
-		require.NoError(err)
-		return len(found) == 2
-	}, 6*time.Second, 100*time.Millisecond)
+	require.False(t, am.Ready())
```

Comment:
What is the purpose of this test, to assert that the Alertmanager is not ready after `setupAMTest`?

Reply:
Yeah, it assets that the construction of the struct was a "success". A newly created AM shouldn't be ready until a configuration is applied to it.